projects/ngneat/error-tailor/src/lib/control-error.directive.ts

```diff
@@ -95,18 +95,19 @@ export class ControlErrorsDirective implements OnInit, OnDestroy {
 
   ngOnDestroy() {
     this.destroy.next();
+    if (this.ref) this.ref.destroy();
   }
 
   private valueChanges() {
     const controlErrors = this.control.errors;
     if (controlErrors) {
       const [firstKey] = Object.keys(controlErrors);
-      const getError = this.globalErrors[firstKey];
-      if (typeof getError !== 'function') {
+      const getError = this.customErrors[firstKey] || this.globalErrors[firstKey];
+      if (!getError) {
         return;
       }
 
-      const text = this.customErrors[firstKey] || getError(controlErrors[firstKey]);
+      const text = typeof getError === 'function' ? getError(controlErrors[firstKey]) : getError;
```

Comment:
Update the docs that we now provide both function and value.

Reply:
Maybe we can change even the getting started useValue showing both use-cases

The general problem that leads me to investigate and correct the code, was that the typing that you use for the errors Partial<Provider> doesn't clear the shape of what you have to provide, that is a Provider<HashMap<string | (err)=>string>> but unfortunately Angular doesn't have a generic Provider<T> if you think that it may be useful I can try to define a more strict generic type , or maybe just document it better may helps 😉

Reply:
We can create our own `Provider` type. Thanks.

Reply:
Do you think that you need to handle only the useValue and useFactory cases or still need someone to provide the errors as a useClass or useExisting ?

Reply:
Only useValue, useFactory, and useClass.

Reply:
Using a class for this is a bit strange for me, to make it work someone need to write a class that contains only public field of type string , or public method that accept one param (err:any) and return string  - am I right? 
Do you think you will use it in such way instead of a factory function to get DI?

Reply:
I know, I just wanted to be flexible. We can start with `useValue` and `useFactory`.

Reply:
I'll do the type refactor in a separate PR

Reply:
Ok, let me know when you've finished with this one.

Reply:
I've just added a commit to clarify the errors use-cases in the readme and ref cleanup 👇